Reflection/ReflectionExtension.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+/**
+ * The <b>ReflectionExtension</b> class reports
+ * information about an extension.
+ *
+ * @link https://php.net/manual/en/class.reflectionextension.php
+ */
+class ReflectionExtension implements Reflector
+{
+    /**
+     * Name of the extension, same as calling the {@see ReflectionExtension::getName()}
+     * method. Read-only, throws {@see ReflectionException} in attempt to write.
```

Comment:
Could you please clarify why you have added doc that property is read-only (provide point in src)? Can't find such information in docs or src.

Reply:
https://3v4l.org/7fTgK

This information is not in the documentation. But I checked that all fields of Reflection classes is readonly (like `name` or `class`).

Reply:
Hm, may be in this case it make sense to define property via `@property-read` in class' PhpDoc? Thus it will be strictly defined as read-only.

Reply:
In addition to the existing?

Reply:
No, instead of existing. Property still will be available in completion and resolve and in the same time it will be resolved as read-only so seems this is better fit for the case.

Reply:
These fields cannot contain an extended description, like`@link https://www.php.net/manual/en/class.reflectionclass.php#reflectionclass.props.name`. Only in this format:

```php
/**
 * @property-read string $name Name of the class.
 */
class ReflectionClass implements Reflector
```

Reply:
So instead of 
```
class ReflectionClass{
public string $name;
}
```
It will be
```
/**
* @property-read $name
*/
class ReflectionClass{}
```
As Example see `EventBuffer::length`

Reply:
In this case read-only property will be highlighted on write attempt 
<img width="312" alt="Screenshot 2020-06-30 at 00 44 16" src="https://user-images.githubusercontent.com/37301285/86059192-efc22800-ba6a-11ea-80f4-a2b9cb504fc2.png">

Reply:
I know, however, I repeat that for such fields it is impossible to add any descriptions except a short one. That is, I should delete all the links to the documentation that were there before.

Reply:
Well, it seems that read-only fields ("name" and "class" only?) don't have really large extended description and we still can add the links as inline tags though. Probably highlighting of potentially failing code is more important then showing several additional lines of docs in this case. And of course I'm talking here only about read-only fields in changed Reflection classes. All other properties should still stay as is.

Reply:
Ok that's quite reasonable: https://github.com/JetBrains/phpstorm-stubs/pull/857/commits/1cc29782e47637cf9b8fee85da35a4f266a43cea